main.cpp

```diff
@@ -94,33 +89,48 @@ struct IntType
     int* intTypeHeap = nullptr;
 }; 
 
+//FloatType definition ///////////////////////////////////////////
 struct FloatType
 {
     FloatType(float);
     ~FloatType();
     operator float() const { return *floatTypeHeap; }
-    FloatType& add(float rhs);
-    FloatType& subtract(float rhs);
-    FloatType& multiply(float rhs);
-    FloatType& divide(float rhs);
+
+    FloatType& operator+=( const float other);
+    FloatType& operator-=(const float other);
+    FloatType& operator/=( const float other);
+    FloatType& operator*=(const float other);
+
+    FloatType& operator=(const FloatType& other); 
+
     FloatType& pow(float power);
     FloatType& pow(const IntType& intRef);
     FloatType& pow(const FloatType& floatRef);
-    FloatType& pow(const DoubleType& doubleRef);    
+    FloatType& pow(const DoubleType& doubleRef);  
+    
 private:
-    float powInternal(float power);
     float* floatTypeHeap = nullptr;
+    float powInternal(float power);
+    
 };
 
+//DoubleType definition /////////////////////////////////////////
+
 struct DoubleType 
 {
     DoubleType(double);
     ~DoubleType();
     operator double() const { return *doubleTypeHeap; }
-    DoubleType& add(double rhs );
-    DoubleType& subtract(double rhs );
-    DoubleType& multiply(double rhs );
-    DoubleType& divide(double rhs );
+
+    DoubleType& operator+=( const double other);
+    DoubleType& operator-=(const double other);
+    DoubleType& operator/=( const double other);
+    DoubleType& operator*=(const double other);
+
+    DoubleType& operator=(const DoubleType& other);
```

Comment:
the Rule of 3/5/0 hasn't been covered yet.  Your `main()` isn't doing any assignment operations so you have no need for this.
- [ ] Remove this `operator=` function.